src/vue-mf-root-config.js

```diff
@@ -1,24 +1,39 @@
 import { registerApplication, start } from "single-spa";
+import {
+  constructApplications,
+  constructRoutes,
+  constructLayoutEngine,
+} from "single-spa-layout";
 
-registerApplication({
-  name: "@vue-mf/navbar",
-  app: () => System.import("@vue-mf/navbar"),
-  activeWhen: "/",
-  customProps: {
+const routes = constructRoutes(document.querySelector("#single-spa-layout"), {
+  props: {
     githubLink: "https://github.com/vue-microfrontends/root-config",
   },
+  loaders: {
+    navbar: "<h1>Loading navbar</h1>",
+  },
+  errors: {
+    navbar: "<h1>Failed to load navbar</h1>",
+  },
 });
 
-registerApplication({
-  name: "@vue-mf/dogs-dashboard",
-  app: () => System.import("@vue-mf/dogs-dashboard"),
-  activeWhen: "/view-doggos",
+const applications = constructApplications({
+  routes,
+  loadApp: ({ name }) => System.import(name),
 });
 
-registerApplication({
-  name: "@vue-mf/rate-dogs",
-  app: () => System.import("@vue-mf/rate-dogs"),
-  activeWhen: "/rate-doggos",
+// Delay starting the layout engine until the styleguide CSS is loaded
+const layoutEngine = constructLayoutEngine({
+  routes,
+  applications,
+  active: false,
+});
+
+applications.forEach(registerApplication);
+
+System.import("@vue-mf/styleguide").then(() => {
+  // Activate the layout engine once the styleguide CSS is loaded
+  layoutEngine.activate();
```

Comment:
We should move the `start()` call on line 39 to here.

Reply:
I tried but for a reason I'm getting a white screen if I do that

![image](https://user-images.githubusercontent.com/5809230/118570090-5ca8e400-b749-11eb-8619-c7a66222700c.png)

Reply:
If you figure it out, we can merge this. I don't have time right now to diagnose it for you